src/main/java/Barry.java

```diff
@@ -0,0 +1,322 @@
+import java.util.Scanner;
+import java.util.ArrayList;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class Barry {
+
+    // A class for elements in the taskList, with a name and an indicator of whether the task is marked
+    private abstract static class Task{
+        private final String taskName;
+        private boolean isDone;
+
+        // Constructor for the Task class, initialised with the name of the task
+        public Task(String name) {
+            this.taskName = name;
+            this.isDone = false;
+        }
+
+        // Allows the user to modify the status of the task
+        public void modifyStatus(boolean status) {
+            this.isDone = status;
+        }
+
+        // Returns the name of the task
+        protected final String getName() {
```

Comment:
Consider making the functions getName and modifyStatus have the same privacy - either both protected or both public.